src/libtiled/tilelayer.cpp

```diff
@@ -594,8 +659,31 @@ TileLayer *TileLayer::clone() const
 TileLayer *TileLayer::initializeClone(TileLayer *clone) const
 {
     Layer::initializeClone(clone);
-    clone->mGrid = mGrid;
+    QHashIterator<QPoint, Chunk* > it(mChunks);
+    while (it.hasNext()) {
+        it.next();
+        for (int x = 0; x < CHUNK_SIZE; ++x) {
+            for (int y = 0;y < CHUNK_SIZE; ++y) {
+                if (!contains(it.key().x() * CHUNK_SIZE + x,
+                              it.key().y() * CHUNK_SIZE + y))
+                    continue;
+
+                clone->setCell(it.key().x() * CHUNK_SIZE + x,
+                               it.key().y() * CHUNK_SIZE + y,
+                               it.value()->cellAt(x, y));
```

Comment:
Okay, I totally forgot that we can access the private members of another object in the same class. Also, forgot about the implicit copy constructor. Nice..!